src/mtcp/mtcp_restart.c

```diff
@@ -668,6 +669,24 @@ static void unmap_memory_areas_and_restore_vdso(RestoreInfo *rinfo)
     }
     MTCP_ASSERT(vdso == rinfo->vdsoStart);
   }
+#else
+    // In commit dec2c26c1eb13eb1c12edfdc9e8e811e4cc0e3c2 , the mremap
+    //   code above was added, and then caused a segfault on restart for
+    //   __i386__ in CentOS 7.  In that case ENABLE_VDSO_CHECK was not defined.
+    //   This version was needed in that commit for __i386__
+    //   (i.e., for multi-arch.sh) to succeed.  But more recently,
+    //   the mremap version seems to succeed even without ENABLE_VDSO_CHECK. 
+    void *vdso = mtcp_sys_mmap(rinfo->vdsoStart,
+                                 vdsoEnd - vdsoStart,
+				 PROT_EXEC | PROT_WRITE | PROT_READ,
+                                 MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
+    mtcp_memcpy(rinfo->vdsoStart, vdsoStart, vdsoEnd - vdsoStart);
```

Comment:
I agree that we are moving further away from the principle of "release often", but releasing broken code is even worse :-). Let's at least document very clearly what's broken and at that point, I won't have any objections to a quick release.